src/client/discord.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+Minetest
+Copyright (C) 2018 Dumbeldor, Vincent Glize <vincent.glize@live.fr>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "client\discord.h"
+#include <ctime>
+#include "discord_rpc.h"
+#include <string>
+#include <assert.h>
+#include <log.h>
+
+std::unique_ptr<Discord> g_discord;
```

Comment:
Again, please initialise discord in the main function or client launcher and *don't* make it a global or singleton or anything which says "static"

Reply:
He already initialize it in **ClientLauncher**.

https://github.com/minetest/minetest/pull/7303/files#diff-fa5188389a6b6ce7d60db5521e104970R166

Reply:
The issue is it being a global, not where it is initialised